packages/runtime/src/httpserver/server.ts

```diff
@@ -0,0 +1,119 @@
+import express, { Application } from 'express'
+import { createServer as createHttpServer, Server as ServerHttp } from 'http'
+import { Server as ServerHttps } from 'https'
+import { initServer, createExpressEndpoints } from '@ts-rest/express'
+import { contract } from '../contract'
+import { ApiInstance } from '../api'
+import { OxigraphInmemBackend } from '../adapter/oxigraph_inmem'
+import { SparqlAdapter, SparqlFetchBackend } from '../adapter'
+import { isLeft } from 'fp-ts/lib/Either'
+import { BackendAdapter } from '../types'
+
+export type ServerConfig = {
+  server: {
+    port: number
+    apiUrl: string
+  }
+  backend:
+    | {
+        type: 'sparqlEndpoint'
+        sparqlEndpoint: {
+          url: string
+        }
+      }
+    | {
+        type: 'memory'
+        memory: {
+          baseIri: string
+        }
+      }
+}
+
+// TODO: rename?
+export class HttpServer {
+  api: ApiInstance
+  app: Application
+  server: ServerHttp | ServerHttps // FIXME : support https
+  config: ServerConfig
+
+  constructor(config: ServerConfig) {
+    this.config = config
+
+    let adapter: BackendAdapter
+    switch (config.backend.type) {
+      case 'memory':
+        adapter = new SparqlAdapter(new OxigraphInmemBackend(config.backend.memory.baseIri))
+        break
+      case 'sparqlEndpoint':
+        adapter = new SparqlAdapter(new SparqlFetchBackend(config.backend.sparqlEndpoint.url))
+        break
+    }
+
+    this.api = new ApiInstance(adapter, new URL(config.server.apiUrl))
+
+    this.app = express()
+
+    // app.use(bodyParser.urlencoded({ extended: false }))
+    // app.use(bodyParser.json())
+
+    const s = initServer()
+
+    const router = s.router(contract, {
+      getInfo: async () => {
+        const info = this.api.get_info()
+
+        return {
+          status: 200,
+          body: info,
+        }
+      },
+      getDocmapById: async (req) => {
+        const iri = decodeURIComponent(req.params.id)
```

Comment:
this URL path parameter is "sometimes" expected to be a URL itself. it is Always expected to be either a URL or composed of allowed URL characters. See the example in integration tests for how this works with a URL as parameter. What is the significance of always decoding? are there cases where a consumer might get bad results because they sent a string that *looks* urlencoded but isn't intended to be?

Reply:
I'm not sure if questions here are rhetorical or something to continue thinking on?